plugins/woocommerce/src/Blocks/Utils/CartCheckoutUtils.php

```diff
@@ -44,6 +44,112 @@ public static function is_checkout_block_default() {
 		return $checkout_page_id && has_block( 'woocommerce/checkout', $checkout_page_id );
 	}
 
+	/**
+	 * Migrate checkout block field visibility attributes to settings.
+	 */
+	protected static function migrate_checkout_block_field_visibility_attributes() {
+		// Before migrating attributes, migrate the "default" options checkout block uses into the settings.
+		update_option( 'woocommerce_checkout_phone_field', 'optional' );
+		update_option( 'woocommerce_checkout_company_field', 'hidden' );
+		update_option( 'woocommerce_checkout_address_2_field', 'optional' );
+
+		// Parse the block from the checkout page.
+		$checkout_blocks = \WC_Blocks_Utils::get_blocks_from_page( 'woocommerce/checkout', 'checkout' );
+
+		if ( empty( $checkout_blocks ) ) {
+			return;
+		}
+
+		$checkout_block = $checkout_blocks[0];
+
+		if ( ! isset( $checkout_block['attrs'] ) ) {
+			return;
+		}
+
+		if ( isset( $checkout_block['attrs']['showPhoneField'], $checkout_block['attrs']['requirePhoneField'] ) ) {
```

Comment:
Are you suggesting we migrate regardless? I was worried we'd be changing site settings if the user hadn't changed the default options. This could impact the shortcodes too. 

I think its only safe to migrate settings if we're sure the merchant has changed these settings, so I would be inclined to change this to:

```
if ( $checkout_block['attrs']['showPhoneField'] ) {
	update_option( 'woocommerce_checkout_phone_field', $checkout_block['attrs']['requirePhoneField'] ? 'required' : 'optional' );
}
```

And leave current defaults alone otherwise. It would mean some stores see a change in fields displayed on the block checkout however.

cc @senadir

Reply:
I've updated the logic to migrate when the attributes are missing. We assume defaults if they are not present in the attributes. See https://github.com/woocommerce/woocommerce/pull/52784/commits/05f411bfc84ec8d31a8e9c11e342325d119b1d14

Reply:
> I've updated the logic to migrate when the attributes are missing. We assume defaults if they are not present in the attributes. See https://github.com/woocommerce/woocommerce/commit/05f411bfc84ec8d31a8e9c11e342325d119b1d14

Yep nice, this is what I meant